mmds/src/data_store.rs

```diff
@@ -227,17 +257,31 @@ mod tests {
         }"#;
 
         let data_store: Value = serde_json::from_str(data).unwrap();
-        // TODO: This should fail; we should only accept String types
-        mmds.put_data(data_store);
-
-        match mmds.get_value("/age".to_string()) {
+        match mmds.put_data(data_store) {
             Ok(_) => assert!(false),
             Err(e) => assert_eq!(e, Error::UnsupportedValueType),
-        };
+        }
+    }
 
-        match mmds.get_value("/age/".to_string()) {
-            Ok(_) => assert!(false),
-            Err(e) => assert_eq!(e, Error::UnsupportedValueType),
-        };
+    #[test]
+    fn test_valid_types() {
```

Comment:
We can remove this test in favour of `test_update_data_store` (see my previous comment).